SetupBLT.cmake

```diff
@@ -66,6 +66,12 @@ if (NOT BLT_LOADED)
     ################################
     include(${BLT_ROOT_DIR}/cmake/BLTOptions.cmake)
 
+    ################################
+    # CMP0057 New is required by blt_add_libraries
```

Comment:
Can you make this comment that it turns on the IN_LIST operator

Reply:
Note:
You won't see this problem if the host project sets the cmake_min_version to a newish version of cmake. 
I suppose it can't hurt to set it in blt as well, but we haven't set any others yet.

Reply:
Yeah. blt/tests/internal/CMakeLists.txt has:
cmake_minimum_required(VERSION 3.1)

This message is actually wrong. It is blt_setup_target() that has the issue.
```
CMake Warning (dev) at /Users/settgast1/Codes/geosx/blt/cmake/BLTPrivateMacros.cmake:139 (if):
  Policy CMP0057 is not set: Support new IN_LIST if() operator.  Run "cmake
  --help-policy CMP0057" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  IN_LIST will be interpreted as an operator when the policy is set to NEW.
  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  /Users/settgast1/Codes/geosx/blt/cmake/BLTMacros.cmake:582 (blt_setup_target)
  src/combine_static_library_test/CMakeLists.txt:33 (blt_add_executable)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at /Users/settgast1/Codes/geosx/blt/cmake/BLTPrivateMacros.cmake:139 (if):
  if given arguments:

    "NOT" "Foo1" "IN_LIST" "_expanded_DEPENDS_ON"

  Unknown arguments specified
Call Stack (most recent call first):
  /Users/settgast1/Codes/geosx/blt/cmake/BLTMacros.cmake:582 (blt_setup_target)
  src/combine_static_library_test/CMakeLists.txt:33 (blt_add_executable)
```

Reply:
given that we don't know exactly which min version to recommend (i am using 3.9, but that might be too stringent), I think it makes sense to set the policy here to avoid headaches